pulp_ansible/app/tasks/collectionversion_index.py

```diff
@@ -0,0 +1,188 @@
+import logging
+from semantic_version import Version
+
+from pulp_ansible.app.models import (
+    AnsibleDistribution,
+    CrossRepositoryCollectionVersionIndex as CVIndex,
+)
+
+from pulpcore.plugin.models import RepositoryVersion
+
+# from pulpcore.plugin.tasking import dispatch
+
+
+log = logging.getLogger(__name__)
+
+
+def _is_indexable(repo, repo_v):
+    # don't bother updating the index with a distribution doesn't exist
+    # if no distro exists, remove the index
+    if not AnsibleDistribution.objects.filter(repository=repo, repository_version=repo_v).exists():
+        CVIndex.objects.filter(repository=repo, repository_version=repo_v).delete()
```

Comment:
I think you should be using an OR Q object here. Distributions do not allow for repository and repository version to be set at the same time.